Automattic/pocket-casts-ios
# Fix playback when switching from iPhone speaker to other audio outputs with trim silence enabled

Fixes #2049 

See original issue for issue description and root cause. Since iOS 17 and WatchOS 10, we receive a notification when the audio route changes due to disconnect. This PR sets the state for `interruptionInProgress` correctly based on the reason of the interruption.

This is complex part of the player state, so I'm submitting a draft PR first for review and discussion. Open to other ideas for resolution.

## To test

1. Play podcast on iPhone speaker
2. Enable Trim Silence
3. Switch to BT headphones (non-Apple)
4. Switch off headphones. Playback should stop
5. Press Play. Playback should start on speaker.
6. Turn on headphones. They will connect to iPhone and playback will stop, instead of continuing on headphones. Player shows pause button, i.e. it thinks it is still playing, but the progress bar is stopped.
7. Apply the PR and repeat steps above. Playback should continue when switching from speaker to headphone and player state should be correct.
8. Test other audio interruptions, e.g. activating Siri, to ensure that the app still handles those interruptions correctly (i.e. pauses playback and resumes when interruption is done).

## Checklist
